GirafIntegrationTest/README.md

```diff
@@ -43,49 +69,165 @@ The tests within this folder are concerned with how all components of the backen
 This allows testing for things like access rights and correct storage and subsequent retrieval from the database. The flipside is that some tests may depend on others(logging in before checking access rights) and that the reason for the failure of a particular test may lie in multiple different components.
 
 ## Testing framework
-Full documentation such as it is can be found at [@test(skip_if_failed=['loginAsKurt'])]. 
+Full documentation can be seen [here](https://docs.python.org/3/library/unittest.html). 
+
+### Naming
+Generally all names should both be in `lower_snake_case`. Exceptions are global constants in `UPPER_SNAKE_CASE` and class names in `UpperCamelCase`.
+
+**All** test methods should be prefixed by `test_` (a method is **only** recognized as a test if it has this prefix) and `domain_`, and decorated by the `order` function, e.g. 
+
+```py
+@order
+def test_account_login_or_something(self):
+    # do your tests
+```
+
+assuming the test is in the `test_account_controller` file. You should also include what your test is doing and which endpoint it is testing in the doc string, e.g.
+
+```py
+@order
+def test_account_login_or_something(self):
+    """
+    Testing <what your method is testing in active tense>  # e.g. "Testing logging in as <user>"
+    
+    Endpoint: <method>:<endpoint>  # e.g. POST:/v1/Account/login
+    """
+    # do your tests
+```
+The first line of the doc string will be printed to terminal when the test is run, everything else is documentation.
+
+### @order
+By default `unittest` runs the test in alphabetical order. To ensure tests are run in the order in which they are written in the code, the default sorter is set to `compare` from `testlib.py`.
+To enforce this, the `@order` decorator **must be appended to all test methods**.
+
+### File/Class setup
+Should you create a new file, you should make sure that it/they follow the same format as the remaining classes, i.e. follow the same naming scheme, `test_domain_....py` for file name and `TestDomain` for the class. 
+
+The class should extend the `GIRAFTestCase` class from `testlib.py`.
+The class should also override the same methods: `setUpClass`, `tearDownClass`, `setUp`, `tearDown`.
```

Comment:
I do not agree. If the test does not need them, there are no point in overwriting them :)

Reply:
Good point. Fixed in bd26b37.